pkg/operation/botanist/component/nginxingress/nginxingress.go

```diff
@@ -234,6 +234,11 @@ func (n *nginxIngress) computeResourcesData() (map[string][]byte, error) {
 					Resources: []string{"leases"},
 					Verbs:     []string{"create"},
 				},
+				{
+					APIGroups: []string{"discovery.k8s.io"},
+					Resources: []string{"endpointslices"},
+					Verbs:     []string{"get", "list", "watch"},
+				},
```

Comment:
It is strange that the same rule is added to the Role and ClusterRole which are bound to the same ServiceAccount. I would expect that there is no need to add it in the Role as the ClusterRole should ensure that nginx-ingress can get/list/watch endpointslices in all namespaces.
Anyways, in the upstream chart it is the same, so we can leave it as it is. Ref https://github.com/kubernetes/ingress-nginx/commit/3579ed04870c77979ec5bf18f4cd00c8763615a1